test/Base_Tests/src/Semantic/Any_Spec.enso

```diff
@@ -38,75 +38,78 @@ add_specs suite_builder =
             Any.x self=Any . to_text . should_equal "Any:Any"
             Any.x self=With_X . should_equal "Any:With_X"
             Any.x self=With_Y . should_equal "Any:With_Y"
-            Any.x (My_Type.Value 22) . should_equal "Any:(My_Type.Value 22)"
-            Any.x (With_X.Value 22) . should_equal "Any:(With_X.Value 22)"
-            Any.x (With_Y.Value 22) . should_equal "Any:With_Y(22)"
-            Date.to_display_text . should_equal "Date"
+            Any.x self=(My_Type.Value 22) . should_equal "Any:(My_Type.Value 22)"
+            Any.x self=(With_X.Value 22) . should_equal "Any:(With_X.Value 22)"
+            Any.x self=(With_Y.Value 22) . should_equal "Any:With_Y(22)"
 
-        group_builder.specify "static method calls on Any should have defaulted self argument to Any" <|
+        group_builder.specify "Static method invocation (==)" <|
             # Any.== is a method that takes two arguments (self other)
-            (Any.==) . should_be_a Function
+            Test.expect_panic_with (Any.==) No_Such_Method
+            Test.expect_panic_with (Any.== Any) No_Such_Method
+            Test.expect_panic_with (Any.== Any Any) No_Such_Method
+            Test.expect_panic_with (Any.== Boolean) No_Such_Method
             (Any.== self=Boolean) . should_be_a Function
             (Any.== self=Any Boolean) . should_be_false
-            (Any.== Any Any) . should_be_true
-            (Any.== Boolean) . should_be_false
-            (Any.== Any) . should_be_true
-            (Any == Boolean) . should_be_false
-            (Any == Any) . should_be_true
+            (Any.== self=Any Any) . should_be_true
 
+        group_builder.specify "Static method invocation (to_text)" <|
             # Any.to_text is a method that takes one argument (self)
-            Any.to_text . should_equal "Any"
+            Test.expect_panic_with (Any.to_text) No_Such_Method
+            Test.expect_panic_with (Any.to_text Any) No_Such_Method
+            Test.expect_panic_with (Any.to_text Vector) No_Such_Method
+            Test.expect_panic_with (Any.to_text [1,2,3]) No_Such_Method
             Any.to_text self=Any . should_equal "Any"
             Any.to_text self=Vector . should_equal "Vector"
             Any.to_text self=[1,2,3] . should_equal "[1, 2, 3]"
-            Any.to_text Any . should_equal "Any"
-            Any.to_text Vector . should_equal "Vector"
-            Any.to_text [1,2,3] . should_equal "[1, 2, 3]"
 
             Vector.to_text . should_equal "Vector"
             Text.to_text . should_equal "Text"
             Number.to_text . should_equal "Number"
             Integer.to_text . should_equal "Integer"
 
+        group_builder.specify "Static method invocation (spec_method)" <|
             # Any.spec_method is a method that takes three arguments (self, arg1, arg2)
-            (Any.spec_method) . should_be_a Function
-            (Any.spec_method Boolean) . should_be_a Function
```

Comment:
As the `spec_method` is an _instance extension_ method of `Any`:
```
Any.spec_method self arg1 arg2 =
    "Any.spec_method:" + "{" + self.to_text + "}{" + arg1.to_text + "}{" + arg2.to_text + "}"
```
then both `Any.spec_method` and `Any.spec_method` are valid, I think.